src/Common/src/TypeSystem/IL/Stubs/ArrayMethodILEmitter.cs

```diff
@@ -97,7 +98,37 @@ private void EmitILForAccessor()
                 }
                 else if (_method.Kind == ArrayMethodKind.Address)
                 {
-                    // TODO: type check - CheckVectorElemAddr
+                    MethodDesc getEETypePtr =
+                        context.GetWellKnownType(WellKnownType.Object).GetKnownMethod("get_EETypePtr", null);
+                    TypeDesc runtimeTypeHandleType = context.GetWellKnownType(WellKnownType.RuntimeTypeHandle);
+                    MethodDesc getValueInternalMethod = runtimeTypeHandleType.GetKnownMethod("GetValueInternal", null);
```

Comment:
I think it would be better to use `EETypePtrOf<T>` here.

GetValueInternal intrinsics works today, but it is fragile because of it depends on IL pattern matching. I think it is likely that we will want to make EETypePtrOf recognized by the JIT as intrinsic at some point to fix this fragility (EETypePtrOf does not require IL pattern matching) and deprecate use of GetValueInternal.